superset-frontend/src/views/CRUD/hooks.ts

```diff
@@ -206,119 +206,128 @@ export function useSingleViewResource<D extends object = any>(
     setState(currentState => ({ ...currentState, ...update }));
   }
 
-  const fetchResource = useCallback((resourceID: number) => {
-    // Set loading state
-    updateState({
-      loading: true,
-    });
+  const fetchResource = useCallback(
+    (resourceID: number) => {
+      // Set loading state
+      updateState({
+        loading: true,
+      });
 
-    return SupersetClient.get({
-      endpoint: `/api/v1/${resourceName}/${resourceID}`,
-    })
-      .then(
-        ({ json = {} }) => {
-          updateState({
-            resource: json.result,
-            error: null,
-          });
-          return json.result;
-        },
-        createErrorHandler(errMsg => {
-          handleErrorMsg(
-            t(
-              'An error occurred while fetching %ss: %s',
-              resourceLabel,
-              JSON.stringify(errMsg),
-            ),
-          );
+      return SupersetClient.get({
+        endpoint: `/api/v1/${resourceName}/${resourceID}`,
+      })
+        .then(
+          ({ json = {} }) => {
+            updateState({
+              resource: json.result,
+              error: null,
+            });
+            return json.result;
+          },
+          createErrorHandler(errMsg => {
+            handleErrorMsg(
+              t(
+                'An error occurred while fetching %ss: %s',
+                resourceLabel,
+                JSON.stringify(errMsg),
+              ),
+            );
 
-          updateState({
-            error: errMsg,
-          });
-        }),
-      )
-      .finally(() => {
-        updateState({ loading: false });
+            updateState({
+              error: errMsg,
+            });
+          }),
+        )
+        .finally(() => {
+          updateState({ loading: false });
+        });
+    },
+    [handleErrorMsg, resourceName, resourceLabel],
```

Comment:
noticed a few cases where errors were not resulting in toasts (after switching between alerts/reports and opening the modals various times). This fixes the error by ensuring we always have a correct reference to the current toast function. 

This is the only change in this block